source/ID/concordium-did.rst

```diff
@@ -0,0 +1,546 @@
+
+Concordium DID Identifier
+=========================
+
+Method Name
+-----------
+
+The Concordium DID method name is ``ccd``.
+A DID identifier that uses this method must begin with the following prefix: ``did:ccd``.
+Per DID specification [DID] this string must be in lowercase.
+
+Identifier Syntax
+-----------------
+
+Concordium DID identifiers are defined by the following ABNF_:
+
+.. code-block:: ABNF
+
+  ccd-did = prefix ":" *1(network ":") (acctype / pkctype / scitype)
+  prefix  = %s"did:ccd"
+  network = "testnet" / "mainnet"
+  acctype = "acc:" 50(base58char)
+  pkctype = "pkc:" 64(base58char)
+  scitype = "sci:" index *1(“:” subindex)
+  index = 1*DIGIT
+  subindex = 1*DIGIT
+  base58char = "1" / "2" / "3" / "4" / "5" / "6" / "7" / "8" / "9" /
+               "A" / "B" / "C" / "D" / "E" / "F" / "G" / "H" / "J" /
+               "K" / "L" / "M" / "N" / "P" / "Q" / "R" / "S" / "T" /
+               "U" / "V" / "W" / "X" / "Y" / "Z" / "a" / "b" / "c" /
+               "d" / "e" / "f" / "g" / "h" / "i" / "j" / "k" / "m" /
+               "n" / "o" / "p" / "q" / "r" / "s" / "t" / "u" / "v" /
+               "w" / "x" / "y" / "z"
+
+.. note::
+    The network part of a CCD DID is optional.
+    *If omitted the network is assumed to be "mainnet" by default*.
+
+.. note::
+    The ``subindex`` part of the smart contract instance ``sci`` is optional.
+    *If omitted the subindex is assumed to be `0` by default.*
+
+Smart Contract Instance Reference Syntax
+----------------------------------------
+
+A DID reference is a DID with additional data, such as query parameters.
+Concordium smart contract entrypoints can be addressed using DID references.
+
+.. code-block:: ABNF
+
+  ccd-sc-ref = "/" entrypoint *1("?" param-name "=" 1*(base58char))
+  param-name = "parameter_json" / "parameter_binary"
+  entrypoint = 1*(ALPHA / DIGIT / punctuation)
+  punctuation = "!" / DQUOTE / "#" / "$" / "%" / "&" / "'" / "(" /
+                ")" / "*" / "+" / "," / "-" / "." / "/" / ";" /
+                "<" / "=" / ">" / "?" / "@" / "[" / "\" / "]" /
+                "^" / "_" / "`" / "{" / "|" / "}" / "~"
+
+See the details about *dereferencing* a smart contract instance reference in :ref:`sci-reference`.
+
+Examples
+--------
+
+An account on testnet:
+
+``did:ccd:testnet:acc:3ZFGxLtnUUSJGW2WqjMh1DDjxyq5rnytCwkSqxFTpsWSFdQnNn``
+
+A smart contract instance on the default network (``mainnet``):
+
+``did:ccd:sci:12:0``
+
+A public key:
+
+``did:ccd:pkc:0c7f4421e44a4385850b883e3bbf098f5a9853ef6f1a862c2ce2856381b5f5e3``
+
+A smart contract instance with the ``getKeys`` entrypoint that does not take any parameters
+
+``did:ccd:sci:321/getKeys``
+
+A smart contract instance with the ``getCredentials`` entrypoint taking a parameter
+
+``did:ccd:sci:123/getCredentials?parameter_json=QWWfHFtjPSLJdrz1beyVPzFmgS2ECbJVe1k2LXPwUGMZRjjpL``
```

Comment:
Do we really want to support JSON for this? What is the value in this as opposed to just the parameter in binary that is passed directly to the contract?

Reply:
Let's use binary data.